lib/baseFactory.js

```diff
@@ -70,11 +70,15 @@ class BaseFactory {
         });
 
         return this.datastore.save(modelConfig)
-            .then(modelData => this.createClass(hoek.applyToDefaults(config, {
-                datastore: this.datastore,
-                scm: this.scm,
-                id: modelData.id
-            })));
+            .then(modelData => {
+                const c = config;
+
+                c.datastore = this.datastore;
```

Comment:
Since `this.datastore`, `this.scm`, and `modelData.id` are always defined, I don't really see the difference.